app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -580,7 +595,15 @@ private void updateMapFragment(boolean isSlightUpdate) {
                         .loadAttractionsFromLocation(curLatLng))
                         .subscribeOn(Schedulers.io())
                         .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe(this::populatePlaces);
+                        .subscribe(this::populatePlaces,
+                                throwable -> {
+                                    if (throwable instanceof UnknownHostException || throwable instanceof ConnectException) {
+                                        showErrorMessage(getString(R.string.slow_internet));
+                                    } else {
+                                        showErrorMessage(throwable.getMessage());
```

Comment:
May be we should replace this with `showErrorMessage(R.string.error_loading_nearby)` with `error_loading_nearby` as string for "Error while loading Nearby locations that need images" or something similar. (I guess conveying the fact "... Nearby locations that need images" in the error would be informative part of the error though not related to the actual error that has occured ;-))

We could put `throwable.getMessage()` where it's appropriate, the log. This way the exception isn't visible to the user.

Alternatively, we could try and find the possible exceptions that happen and provide a more user-friendly message for each exception. This should improve the user experience. Anyway, showing the exception message to the user would in no way be helpful to the non-technical users.